agent/src/agent.php

```diff
@@ -76,6 +77,14 @@
 $debug = in_array($_SERVER['NIGHTWATCH_DEBUG'] ?? null, ['true', '1'], true);
 /** @var ?string $basePath */
 $basePath ??= str_replace(['phar://', '/agent.phar/src'], '', __DIR__);
+/** @var string|false $expectedSignature */
+$expectedSignature = file_get_contents($basePath.'/signature.txt');
+if ($expectedSignature === false) {
+    $error("Unable to read the agent's signature");
+
+    return;
+}
+/** @var string $expectedSignature */
```

Comment:
We shouldn't need this one. PHP tooling can tell that this is only now possibly a string, as the previous conditional returns when false.